### ISSUE TITLE
EPOS particle history missing

### ISSUE THREAD

**Issue by HDembinski** [impy-project/impy#53]
While working on the HepMCWriter, I noticed that the EposLHC record contains invalid indices in the parents that go beyond the range of particles. This needs some investigation and fixes.

**Comment by afedynitch** [comment]
Aren't the parent indices for EPOS not simply zero??

```
for event in epos(10):
    nhep = epos.lib.hepevt.nhep
    for i in range(nhep):
        print(i, epos.lib.hepevt.idhep[i], epos.lib.hepevt.isthep[i], epos.lib.hepevt.jmohep[:,i])
```
Gives:
```
0 2212 4 [-1 -1]
1 2212 4 [-1 -1]
2 -211 1 [0 0]
3 111 1 [0 0]
4 111 1 [0 0]
5 -211 1 [0 0]
6 211 1 [0 0]
7 -321 1 [0 0]
8 111 1 [0 0]
9 -211 1 [0 0]
10 111 1 [0 0]
11 111 1 [0 0]
12 111 1 [0 0]
13 -211 1 [0 0]
14 2212 1 [0 0]
15 -211 1 [0 0]
16 111 1 [0 0]
17 211 1 [0 0]
18 211 1 [0 0]
19 211 1 [0 0]
20 -211 1 [0 0]
21 -2112 1 [0 0]
22 211 1 [0 0]
```
etc...

**Comment by HDembinski** [comment]
I haven't looked into the details yet, but now that I see this, I guess that the code gets confused by `[-1, -1]`. Negative indices do not happen in the Pythia convention.

**Comment by HDembinski** [comment]
EPOS does have particle history, though, doesn't it? Can you also activate this in Fortran like you did in SIBYLL somehow?

**Comment by afedynitch** [comment]
I will ask Tanguy if there is any chance to get this out of EPOS. There should be some history for sure!

**Comment by HDembinski** [comment]
I fixed the main issue in the EPOSEvent in #54, by changing the parents of beam particles to (0, 0) and by making all other particles daughters of the beam particles.

So what's left is to enable decay history.
